r/vignettes/read_write.Rmd

```diff
@@ -0,0 +1,164 @@
+---
+title: "Reading and writing data files"
+description: >
+  Learn how to read and write CSV, Parquet, and Feather files with `arrow` 
+output: rmarkdown::html_vignette
+---
+
+The `arrow` package provides functions for reading single data files in
+several common formats. By default, calling any of these functions
+returns an R data frame. To return an Arrow Table, set argument
+`as_data_frame = FALSE`.
```

Comment:
What do you think to the idea of mentioning in this paragraph that the file reader functions read things into memory?  I've been in a few discussions lately where it's been mentioned that the eager/lazy distinction for the `read_*()` functions as compared to `open_dataset()` is a more accurate one than single/multi-files (you can read in a single file via the datasets API), and whilst we haven't started pushing that narrative yet, here could be a good place to start?

Reply:
That's nice, I'll do that. @stephhazlitt had some suggestions about other places we can rephrase to emphasise the in-memory/on-disk distinction too

Reply:
Great!  Steph mentioned earlier that you'd coined "in-memory vs. larger-than-memory" btw; I really like that!